apps/studio/components/interfaces/Organization/BillingSettings/Subscription/SubscriptionPlanUpdateDialog.tsx

```diff
@@ -0,0 +1,644 @@
+                  : PLAN_HEADINGS[(selectedTier as PlanHeadingKey) || 'default']}
+              </h3>
+
+              {subscriptionPreviewIsLoading && (
+                <div className="space-y-2 mb-4 mt-2">
+                  <ShimmeringLoader />
+                  <ShimmeringLoader className="w-3/4" />
+                  <ShimmeringLoader className="w-1/2" />
+                </div>
+              )}
+              {subscriptionPreviewInitialized && (
+                <>
+                  <Table className="mt-2 mb-4 text-foreground-light">
+                    <TableBody>
+                      {(() => {
+                        // Calculate remaining days in current billing cycle
+                        const now = Math.floor(Date.now() / 1000) // current time in seconds
+                        const remainingSeconds = subscription?.current_period_end - now
+                        const totalSeconds =
+                          subscription?.current_period_end - subscription?.current_period_start
+                        const remainingRatio = remainingSeconds / totalSeconds
+
+                        // Calculate prorated credit for current plan
+                        const currentPlanMonthlyPrice = currentPlanMeta?.price ?? 0
+                        const proratedCredit = currentPlanMonthlyPrice * remainingRatio
+
+                        // Calculate new plan cost
+                        const newPlanCost = subscriptionPlanMeta?.priceMonthly ?? 0
+
+                        const customerBalance = ((subscription?.customer_balance ?? 0) / 100) * -1
+
+                        // Calculate total charge (new plan - prorated credit)
+                        const totalCharge = Math.max(
+                          0,
+                          newPlanCost - proratedCredit - customerBalance
+                        )
+
+                        return (
+                          <>
+                            <TableRow>
+                              <TableCell className="py-2 pl-0 flex items-center gap-1">
+                                <span>{subscriptionPlanMeta?.name} Plan</span>
+                                <Badge variant={'brand'} size={'small'} className="ml-1">
+                                  New
+                                </Badge>
+                              </TableCell>
+                              <TableCell className="py-2 pr-0 text-right">
+                                {formatCurrency(newPlanCost)}
+                              </TableCell>
+                            </TableRow>
+                            {subscription?.plan?.id !== 'free' && (
+                              <TableRow>
+                                <TableCell className="py-2 pl-0 flex items-center gap-1">
+                                  <span>Unused Time on {subscription?.plan?.name} Plan</span>
+                                  <InfoTooltip className="max-w-sm">
```

Comment:
Small nit here - somehow the first tooltip is opened the moment the dialog is open
Likewise for when downgrading to Pro

https://github.com/user-attachments/assets/3524d1e1-7552-488e-80e5-6cd7eddef1af

Reply:
fixed